solr/core/src/java/org/apache/solr/schema/IndexSchema.java

```diff
@@ -1225,6 +1227,16 @@ public SchemaField getFieldOrNull(String fieldName) {
     return f;
   }
 
+  public SchemaField getFieldOrNull(SortOrder sortOrder) {
+    if (sortOrder instanceof SortField) {
+      String field = sortOrder.name();
```

Comment:
Couldn't you cast to `SortField` and use `.getField()`?

I think this `name`/`field`/`toString` triality is confusing...